job.c

```diff
@@ -12,7 +12,7 @@ static job *all_jobs = all_jobs_init;
 static size_t all_jobs_cap = 12289; /* == primes[0] */
 static size_t all_jobs_used = 0;
 
-static int hash_table_was_oom = 0;
+static int hash_table_was_oom = 1;
```

Comment:
So close! Thank you for doing all these revisions. :)

We just need to initialize this variable to 0 here. (If it's initialized to 1, the hash table will never grow.)